CLI: Implement `auth logout` command

fredrikekelund opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "Fixes" keyword and use "Related to" instead.
-->

## Related issues

- Fixes STU-947

## Proposed Changes

This PR is based on @youknowriad's work in #1958. It implements a `studio auth logout` command that:

1. Calls a REST API endpoint to revoke the access token
2. Clears the access token from the appdata file

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1. Open Studio
2. Ensure you are logged out
3. Run `npm run cli:build`
4. Run `node dist/cli/main.js auth logout`
5. Ensure that an "Already logged out" message is printed
6. Log in to WordPress.com through Studio
7. Run `node dist/cli/main.js auth logout`
8. Ensure that a "Successfully logged out" message is printed
9. Reload Studio
10. Ensure that you are logged out in Studio, too

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Have you checked for TypeScript, React or other console errors?

fredrikekelund committed:
CLI: Implement `auth login` command

fredrikekelund committed:
Fix lint and tests

fredrikekelund committed:
Add tests, tweak command

fredrikekelund committed:
Consider expiration time

fredrikekelund committed:
Move logic around

fredrikekelund committed:
Unwrap

fredrikekelund committed:
Fix type error

fredrikekelund committed:
CLI: Implement `auth logout` command

fredrikekelund committed:
Address review comments

fredrikekelund committed:
Reject unsupported platforms

fredrikekelund committed:
Merge branch 'f26d/cli-auth-login-command' into f26d/cli-auth-logout-command

fredrikekelund committed:
Merge branch 'trunk' into f26d/cli-auth-login-command

fredrikekelund committed:
Address review feedback

fredrikekelund committed:
Fix test

fredrikekelund committed:
Merge branch 'f26d/cli-auth-login-command' into f26d/cli-auth-logout-command

fredrikekelund committed:
Fix

gcsecsey reviewed (approved):
Thanks @fredrikekelund, this looks good to me and works well. 👍 

The command's output is as described:

| Logged in | Logged out |
|--------|--------|
| <img width="768" height="216" alt="Image" src="https://github.com/user-attachments/assets/fa19f442-2f0f-46f7-9383-6df2789bea07" /> | <img width="734" height="340" alt="Image" src="https://github.com/user-attachments/assets/5261b82a-8c8f-4276-a6fb-02553c2b0b9f" /> |

When switching back to Studio, I'm also logged out:

<img width="2114" height="1442" alt="Image" src="https://github.com/user-attachments/assets/cd252426-a5e8-480f-8406-9e2271b90b03" />

The new tests are also all green:

<img width="700" height="509" alt="Image" src="https://github.com/user-attachments/assets/afa89fc0-445f-4d9f-9020-c27f5208ca04" />

fredrikekelund committed:
Merge branch 'trunk' into f26d/cli-auth-login-command

github-actions[bot] committed:
docs: Update CLI documentation for auth login command

Add comprehensive documentation for the new `studio auth login` command
including usage instructions, flow description, and important notes about
token lifetime and authentication sharing between CLI and desktop app.

Co-authored-by: Fredrik Rombach Ekelund <undefined@users.noreply.github.com>

fredrikekelund committed:
Restore package-lock.json

fredrikekelund committed:
Install inquirer dependency again

fredrikekelund committed:
Tweak

fredrikekelund committed:
Merge branch 'f26d/cli-auth-login-command' into f26d/cli-auth-logout-command

fredrikekelund committed:
Merge branch 'trunk' into f26d/cli-auth-logout-command

fredrikekelund committed:
Fix merge

github-actions[bot] committed:
docs: Update CLI documentation

Add documentation for the new `studio auth logout` command, following the same format as the existing `auth login` command documentation.

Co-authored-by: Fredrik Rombach Ekelund <undefined@users.noreply.github.com>

bcotrim commented on cli/commands/auth/logout.ts:
```diff
@@ -0,0 +1,55 @@
+	}
+
+	try {
+		await revokeAuthToken( token.accessToken );
```
In Studio, failing to revoke the auth token doesn't fail the logout operation.
Have you considered keeping the same behavior in the CLI?

bcotrim replied:
If we consider the revoke of the token as non-critical we should do the api call after removing the token from appdata, preventing the unlikely scenario of having a revoked token store in appdata

fredrikekelund committed:
Tweaks

bcotrim commented on cli/commands/auth/logout.ts:
```diff
@@ -0,0 +1,55 @@
+	try {
+		await revokeAuthToken( token.accessToken );
+
+		await lockAppdata();
```
It's unlikely, but I think it's a good approach.

fredrikekelund replied:
I agree the case that Claude describes is very unlikely, but I still realize that `lockAppdata` should always be the first thing we call in a `try..catch` block to prevent cases where we unlock appdata without actually having acquired the lock.

bcotrim reviewed (approved):
LGTM 👍 

Tested and worked as described on macOS and Windows.
Added some non-blocking considerations

fredrikekelund committed:
Lock appfile before calling API

fredrikekelund committed:
Fix test
